redfish-core/lib/processor.hpp

```diff
@@ -851,6 +901,318 @@ inline void getProcessorData(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+
+    // Get Core object dbus service name
+    crow::connections::systemBus->async_method_call(
+        [objPath, aResp{std::move(aResp)}](const boost::system::error_code ec,
+                                           const GetObjectType& objType) {
+            if (ec || objType.empty())
+            {
+                BMCWEB_LOG_ERROR << "DBUS response error [" << ec.value()
+                                 << " : " << ec.message()
+                                 << "] when tried to get the dbus name for the "
+                                    "given object: "
+                                 << objPath;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            if (objType.size() > 1)
+            {
+                BMCWEB_LOG_ERROR << "More than one dbus service implemented "
+                                    "the object: "
+                                 << objPath;
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            if (objType[0].first.empty())
+            {
+                BMCWEB_LOG_ERROR << "The retrieved dbus name of the given "
+                                 << "object [" << objPath << "] is empty";
+                messages::internalError(aResp->res);
+                return;
+            }
+
+            crow::connections::systemBus->async_method_call(
+                [objPath,
+                 aResp](const boost::system::error_code ec,
+                        const dbus::utility::ManagedObjectType& dbusData) {
+                    if (ec)
+                    {
+                        BMCWEB_LOG_DEBUG << "DBUS response error, ec: "
+                                         << ec.value();
+                        messages::internalError(aResp->res);
+                        return;
+                    }
+
+                    for (const auto& [path, interfaces] : dbusData)
+                    {
+                        if (path != objPath)
+                        {
+                            continue;
+                        }
+
+                        bool present = false;
+                        bool functional = false;
+
+                        for (const auto& [interface, properties] : interfaces)
+                        {
+                            if (interface == "xyz.openbmc_project.State."
+                                             "Decorator.OperationalStatus")
+                            {
+                                for (const auto& [proName, proValue] :
+                                     properties)
+                                {
+                                    if (proName == "Functional")
+                                    {
+                                        const bool* value =
+                                            std::get_if<bool>(&proValue);
+                                        if (value != nullptr)
```

Comment:
Let's turn this around and do 

                                         if (value == nullptr)
                                         {
                                         
                                         
                                         }

Reply:
Let's do this other places as well